src/Utils/Startup/windows/WindowsScriptManager.ts

```diff
@@ -0,0 +1,63 @@
+import { ScriptManager } from '../SсriptManager';
+import { StartupOptions } from '../StartupOptions';
+import { PidStore } from '../PidStore';
+import { RegistryHelper } from './RegistryHelper';
+import WinReg from 'winreg';
+import { ChildProcess, spawn } from 'child_process';
+
+export class WindowsScriptManager implements ScriptManager {
+  private readonly _pidStore;
+
+  constructor(pidStore: PidStore) {
+    this._pidStore = pidStore;
+  }
+
+  public getPidIfExists(): Promise<number | undefined> {
+    return this._pidStore.get() || undefined;
+  }
+
+  public install(opts: StartupOptions): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const autoRunKey = RegistryHelper.getRunKey();
+      autoRunKey.set(
+        opts.name,
+        WinReg.REG_SZ,
+        this.getRunString(opts),
+        (error) => {
+          if (error) {
+            reject(error);
+          }
+          resolve();
+        }
+      );
+    });
+  }
+
+  public uninstall(name: string): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      const autoRunKey = RegistryHelper.getRunKey();
+      autoRunKey.remove(name, (err) => {
+        if (err) {
+          reject(err);
+        }
```

Comment:
You have to break execution in such cases
```suggestion
        if (err) {
          return reject(err);
        }
```